metrics/metrics-integration/metrics-overwatch/src/main/java/org/apache/servicecomb/metrics/overwatch/OverwatchMetricsPublisher.java

```diff
@@ -0,0 +1,136 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.servicecomb.metrics.overwatch;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.servicecomb.foundation.common.net.NetUtils;
+import org.apache.servicecomb.foundation.common.utils.JsonUtils;
+import org.apache.servicecomb.metrics.common.ConsumerInvocationMetric;
+import org.apache.servicecomb.metrics.common.HealthCheckResult;
+import org.apache.servicecomb.metrics.common.MetricsDimension;
+import org.apache.servicecomb.metrics.common.RegistryMetric;
+import org.apache.servicecomb.metrics.core.publish.HealthCheckerManager;
+import org.apache.servicecomb.metrics.overwatch.dto.InstanceStatus;
+import org.apache.servicecomb.metrics.overwatch.dto.SystemFailure;
+import org.apache.servicecomb.metrics.overwatch.dto.SystemStatus;
+import org.apache.servicecomb.metrics.push.MetricsPusher;
+import org.apache.servicecomb.provider.springmvc.reference.RestTemplateBuilder;
+import org.apache.servicecomb.serviceregistry.RegistryUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.netflix.config.DynamicPropertyFactory;
+
+@Component
+public class OverwatchMetricsPublisher implements MetricsPusher {
+  private static final Logger logger = LoggerFactory.getLogger(OverwatchMetricsPublisher.class);
+
+  private static final String METRICS_OVERWATCH_ADDRESS = "servicecomb.metrics.overwatch.address";
+
+  private static final String METRICS_OVERWATCH_WINDOW_TIME = "servicecomb.metrics.overwatch.window_time";
+
+  private final String overwatchURL;
+
+  private final String overwatchFailureURL;
+
+  private final RestTemplate template;
+
+  private final HealthCheckerManager checkerManager;
+
+  public OverwatchMetricsPublisher(HealthCheckerManager checkerManager) {
+    this.checkerManager = checkerManager;
+    String url = "http://" + DynamicPropertyFactory.getInstance()
```

Comment:
publisher should be full customize by developers, including schema(https), and certificates etc.

Reply:
DONE